01/lib/raii_action.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <concepts>
+
+
+template <typename T>
+concept callable =
+    requires(T func)
+    {
+        func();
+    };
+
+template <callable T>
+class RAIIAction {
+private:
+    T func;
+
+public:
+    RAIIAction(T func_) : func(func_) {}
```

Comment:
соответственно в конструкторе 
```suggestion
    RAIIAction(T func) : func_(func) {}
```